src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,11 +1,34 @@
-export const TodoFilter = () => (
+import React from 'react';
+
+import { FilterType, FilterOptions } from '../../types/FilterType';
+
+type Props = {
+  filter: FilterType,
+  onFilterSet: (event:
+  React.ChangeEvent<HTMLSelectElement | HTMLInputElement>) => void,
+  onInputClear: () => void,
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  filter,
+  onFilterSet,
+  onInputClear,
+}) => (
   <form className="field has-addons">
     <p className="control">
       <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
+        <select
+          data-cy="statusSelect"
+          name="filterByStatus"
+          value={filter.filterByStatus}
+          onChange={onFilterSet}
+        >
+          {(Object.keys(FilterOptions) as (keyof typeof FilterOptions)[])
+            .map((key) => (
+              <option value={FilterOptions[key]} key={key}>
+                {key}
+              </option>
+            ))}
```

Comment:
```suggestion
          {(Object.entries(FilterOptions))
            .map(([key, value]) => (
              <option value={value} key={key}>
                {key}
              </option>
            ))}
```

Reply:
Will it work the same?

Reply:
Thank you very much! I have already used the construction in the TodoApp task. The very efficient way to work with enum